plugins/modules/ipaconfig.py

```diff
@@ -415,38 +414,33 @@ def main():
                     except ipalib_errors.EmptyModlist:
                         changed = False
         else:
-            rawresult = ansible_module.ipa_command_no_name("config_show", {})
-            result = rawresult['result']
             del result['dn']
+            type_map = {"str": str, "int": int, "list": list, "bool": bool}
             for key, value in result.items():
                 k = reverse_field_map.get(key, key)
                 if ansible_module.argument_spec.get(k):
-                    if k == 'ipaselinuxusermaporder':
-                        exit_args['ipaselinuxusermaporder'] = \
-                            result.get(key)[0].split('$')
-                    elif k == 'domain_resolution_order':
-                        exit_args['domain_resolution_order'] = \
-                           result.get(key)[0].split('$')
-                    elif k == 'usersearch':
-                        exit_args['usersearch'] = \
-                            result.get(key)[0].split(',')
-                    elif k == 'groupsearch':
-                        exit_args['groupsearch'] = \
-                            result.get(key)[0].split(',')
-                    elif isinstance(value, str) and \
-                            ansible_module.argument_spec[k]['type'] == "list":
+                    arg_type = ansible_module.argument_spec[k]['type']
+                    if k in (
+                           'ipaselinuxusermaporder', 'domain_resolution_order'
+                       ):
+                        exit_args[k] = result.get(key)[0].split('$')
+                    elif k in (
+                           'usersearch', 'groupsearch'
+                       ):
+                        exit_args[k] = result.get(key)[0].split(',')
+                    elif isinstance(value, str) and arg_type == "list":
                         exit_args[k] = [value]
-                    elif isinstance(value, list) and \
-                            ansible_module.argument_spec[k]['type'] == "str":
-                        exit_args[k] = ",".join(value)
-                    elif isinstance(value, list) and \
-                            ansible_module.argument_spec[k]['type'] == "int":
-                        exit_args[k] = ",".join(value)
-                    elif isinstance(value, list) and \
-                            ansible_module.argument_spec[k]['type'] == "bool":
+                    elif (
+                        isinstance(value, (tuple, list))
+                        and arg_type in ("str", "int")
+                    ):
+                        exit_args[k] = type_map[arg_type](value[0])
+                    elif (
+                        isinstance(value, (tuple, list)) and arg_type == "bool"
+                    ):
                         exit_args[k] = (value[0] == "TRUE")
                     else:
-                        exit_args[k] = value
+                        exit_args[k] = type_map[arg_type](value)
```

Comment:
Which values can `arg_type` have?

Reply:
The valid values are `str`, `int`, `list` and `bool`. In practice, at this point, `bool` will not occur.

Reply:
I think it would be good to have an additional arg_type check to make sure that there is not another type used. It would lead to a backtrace here.

Reply:
Added da test that will raise an ValueError if `arg_type` is not found in `type_map`. This should be show on tests if any new data type is added to the argument spec for the module.